### ISSUE TITLE
a ‘pass’ statement in an intentionally empty function with a docstring is not useless

### ISSUE THREAD

**Issue by wbolster** [PyCQA/autoflake#73]
not all `pass` statements occurring in places where they are not _technically required_ are useless.

consider for instance a command line app written using the popular `click` library, featuring some subcommands ([see docs](https://click.palletsprojects.com/en/7.x/commands/)). for the sake of argument, consider a hypothetical `git`-like application supporting a command like `git remote add`, which would look like this:

``` python
@click.group()
def remote_group(...):
    pass

@remote_group.command():
def add(...):
    ...  # implementation goes here
```

now, let's add docstrings, which in the case of `click`, also appear in `--help` output:

``` python
@click.group()
def remote_group(...):
    """
    Manage set of tracked repositories.
    """
    pass

@remote_group.command()
def add(...):
    """
    Add a new remote.
    """
    ...  # implementation goes here
```

technically, the presence of a docstring means the `pass` is not needed to make it valid python. but the `pass` shows the intent to the reader. without it i would immediately wonder if someone forget to write some code, or if some git merge went wrong. in this specific example, `click` actually supports shared code in functions decorated with `click.group()`. the presence of `pass` signals that this function does not need that, and is _intended_ as an empty body.

so, in short: not all `pass` statements that are _strictly technically unnecessary_ are to be considered ‘useless’.

my suggestion would be to treat the ‘function body, with only a docstring and a pass statement’ as an extra special case, and _always_ leave the `pass` statements in place.

**Comment by grthr** [comment]
I just came across this tool and was very exited. But this is a showstopper.

Can you make it configurable to remove 'useless' `pass` statements or not?

**Comment by kamalmarhubi** [comment]
This also happens if you use ABCs with methods decorated with `@abstractmethod`: they will typically have a docstring but no implementation.

**Comment by EdvardM** [comment]
Sometimes people use ellipsis `...` in empty functions to mark it as implementation missing, though not sure is that more of a hack, as it has varying semantics depending on where it's used. 

Still, I actually prefer using ellipsis in these cases, as it stands out more clear that method/function doesn't have implementation, and autoflake doesn't seem to complain about it either.

**Comment by wbolster** [comment]
`...` reads as a placeholder for code that needs to be written, or that it's a ‘template’, e.g. in `.pyi` stub files.

`pass` in an exception class body means it't _intentionally empty_, see e.g. the `click` example and its rationale in my earlier comment.

**Comment by EdvardM** [comment]
Good explanation, thanks!

**Comment by PhilippSelenium** [comment]
Will this be fixed?
